Self-Serve Stork

ACK101101 opened the pull request.

ACK101101 commented on chains/evm/contracts/first_party_stork/contracts/FirstPartyStork.sol:
am not sure which functions should be here besides updateTemporalNumericValues

ACK101101 commented on chains/evm/contracts/first_party_stork/contracts/UpgradeableFirstPartyStork.t.sol:
this is still a work in progress

ACK101101 commented on chains/evm/contracts/first_party_stork/ignition/modules/FirstPartyStork.ts:
this is used for deploying the contracts with `npx hardhat ignition deploy ignition/modules/FirstPartyStork.ts --network localhost`

ACK101101 commented on chains/evm/contracts/first_party_stork/scripts/local_register_publisher.ts:
this is called in the docker entrypoint script - it registers the user's PUBLISHER_EVM_PUBLIC_KEY as a valid publisher for the local contract

ACK101101 commented on chains/evm/contracts/first_party_stork/README.md:
this is a placeholder

ACK101101 commented on chains/evm/contracts/first_party_stork/test_viem.js:
this is temporary

ACK101101 commented on docker/scripts/first-party-evm-docker-entrypoint.sh:
```diff
@@ -0,0 +1,20 @@
+npx wait-on http://127.0.0.1:8545 && npx hardhat ignition deploy ignition/modules/FirstPartyStork.ts --network hardhatLocal --reset
+
+# Run the TypeScript registration script and log success
+npx ts-node scripts/local_register_publisher.ts && echo "REGISTRATION_COMPLETE" >> /tmp/registered.log
```
this is where the register script is run, it writes to a tmp file that the docker-compose healthcheck checks for. this prevents the startup race between the pusher and the contract.
the temp file is cleared out before each run on line 7

ACK101101 commented on scripts/animate.sh:
```diff
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-STARTUP_ANIMATION_PATH="apps/lib/generate/frames"
+STARTUP_ANIMATION_PATH="utils/generate/pkg/frames"
```
this was broken

ACK101101 commented on shared/signer/model.go:
moved to shared

ACK101101 commented on shared/errors.go:
maybe overkill but i could see this being helpful in the future

ACK101101 commented on Makefile:
fixed these commands
`| cut -d: -f1` works if there are multiple GOPATH's

HenryMBaldwin commented on apps/first_party_pusher/docker-compose.yml:
Could this be a profile in the root docker compose/ go into docker/compose/ or something?

HenryMBaldwin commented on apps/first_party_pusher/README.md:
```diff
@@ -0,0 +1,147 @@
+
+## Development
+
+### Adding New Chain Support
```
The "Adding New Chain Support" section seems both wrong and unnecessary.

HenryMBaldwin commented on chains/evm/sdks/first_party_stork_evm_sdk/README.md:
```diff
@@ -0,0 +1,190 @@
+
+## Overview
+
+The FirstParty Stork EVM SDK allows developers to interact with FirstParty Stork oracle contracts, which enable publishers to directly submit price updates on-chain. Unlike the standard Stork oracle that uses centralized aggregation, FirstParty Stork allows individual publishers to submit their own price data with configurable fees and historical data storage.
```
Though this is correct, it might be nice to avoid publicly calling ourselves centralized - noted that this is likely a placeholder still just shouting it out.

akawalsky commented on chains/evm/contracts/first_party_stork/foundry.toml:
```diff
@@ -0,0 +1,7 @@
+[profile.default]
```
Agreed here

akawalsky reviewed (commented):
A few initial comments. Looking good, let me know when its ready for re-review after adding more tests and updating the readmes

harryrackmil commented on apps/publisher_agent/pkg/runner.go:
```diff
@@ -72,40 +85,56 @@ func (r *PublisherAgentRunner[T]) UpdateBrokerConnections() {
-		return
+	}
+
+	// merge seeded brokers with registry brokers
```
have we tested the case where we have both a registry for a url and seeded brokers for that url? Logic seems solid but worth a unit test

harryrackmil commented on apps/publisher_agent/pkg/registry.go:
```diff
@@ -24,18 +25,20 @@ func NewRegistryClient(baseUrl string, storkAuthSigner signer.StorkAuthSigner, l
-) (map[BrokerPublishUrl]map[AssetId]struct{}, error) {
+	publisherKey shared.PublisherKey,
+) (map[BrokerPublishUrl]map[shared.AssetID]struct{}, error) {
+	brokerMap := make(map[BrokerPublishUrl]map[shared.AssetID]struct{})
```
hmm I kinda prefer the current code? `return nil, error` seems pretty idiomatic in go so I'd kinda prefer to return nil rather than an empty map in case a user checked the return value for some reason. Eyeballing the code it's not obvious from `return brokerMap, fmt.Errorf("not authorized to query Stork Registry - check your configured StorkAuth")` that we're not returning a meaningful map.

Note that there are some cases where it might make sense to return a real (non-empty) map with an error (e.g. we had an error along the way but here's all the brokers that succeeded) so it's not always the case that the return object doesn't matter just because we're returning an error.

ACK101101 commented on apps/first_party_pusher/docker-compose.yml:
```diff
@@ -0,0 +1,151 @@
+      evm-contract:
+        condition: service_healthy
+
+  first-party-pusher-production:
```
i guess it isnt hard for users to just run the pusher and agent by themselves if they want - ill remove this concept

ACK101101 commented on chains/evm/contracts/first_party_stork/hardhat.config.ts:
```diff
@@ -0,0 +1,56 @@
+    },
+  },
+  networks: {
+    localhost: {
```
removed localhost

ACK101101 commented on chains/evm/contracts/first_party_stork/hardhat.config.ts:
```diff
@@ -0,0 +1,56 @@
+      type: "http",
+      chainType: "l1",
+      url: configVariable("SEPOLIA_RPC_URL"),
+      accounts: [configVariable("SEPOLIA_PRIVATE_KEY")],
```
Gotcha, removing for now since it isn't being used and was more of a placeholder when I was first creating the hardhat config

ACK101101 commented:
Addressed comments. It seems like the check-links test is a bit flakey. Anecdotally, `https://www.npmjs.com/package/@storknetwork/stork_pyth_adapter` seems to be the most problematic, but I have seen the test fail for a bunch of our different npm packages.
